src/Microsoft.AspNetCore.SystemWebAdapters.SessionState/RemoteSession/RemoteAppSessionStateManager.cs

```diff
@@ -23,19 +23,21 @@ internal partial class RemoteAppSessionStateManager : ISessionManager
     private readonly RemoteAppSessionStateClientOptions _options;
 
     public RemoteAppSessionStateManager(
-        HttpClient client,
+        IHttpClientFactory httpClientFactory,
         ISessionSerializer serializer,
         IOptions<RemoteAppSessionStateClientOptions> sessionOptions,
         IOptions<RemoteAppOptions> remoteAppOptions,
         ILogger<RemoteAppSessionStateManager> logger)
     {
-        _client = client ?? throw new ArgumentNullException(nameof(client));
-        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        var remoteOptions = remoteAppOptions?.Value ?? throw new ArgumentNullException(nameof(remoteAppOptions));
```

Comment:
do we need to get the value initially? That forces all the callbacks to be run initially at constructor injection rather than deferring to when (or if) it is actually needed

Reply:
We can wait until the HttpClient is needed. I'll update it to have a Client property which sets this up lazily.